vm-setup/roles/v1aX_integration_test/filter_plugins/status.py

```diff
@@ -0,0 +1,70 @@
+""" filtering k8s_info resources """
+
+from ansible.errors import AnsibleFilterError
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def msg(key, resources):
+    return "could not find %r key in %r" % (key, resources)
+
+
+def filter_phase(resources, phase, skip_missing=True):
+    """return resources in a defined phase"""
+    filtered = []
+    for r in resources:
+        try:
+            if r["status"]["phase"].lower() == phase:
+                filtered.append(r)
+        except KeyError:
+            if skip_missing:
+                display.warning(msg("['status']['phase']", resources))
+                break
+            raise AnsibleFilterError(msg("['status']['phase']", resources))
+
+    return filtered
+
+
+def filter_ready(resources, ready=True, skip_missing=True):
+    """return resources based on defined ready status"""
+    filtered = []
+    for r in resources:
+        try:
+            if r["status"]["ready"] == ready:
+                filtered.append(r)
+        except KeyError:
+            if skip_missing:
+                display.warning(msg("['status']['ready']", resources))
+                break
+            raise AnsibleFilterError(msg("['status']['ready']", resources))
+
+    return filtered
+
+
+def filter_provisioning(resources, state, skip_missing=True):
+    """return resources in a defined provisioning state"""
```

Comment:
It's not clear to me what I would need to pass as `state` to this filter to use properly. Instead, could we call the filter `filter_provisioned` which only takes in resources as an argument and returns those which are provisioned? This would be very clear to me.

Reply:
Same as phase, we will end up having `filter_provisioning_provisioned `and `filter_provisioning_ready ` with duplicated code

Check details: https://github.com/metal3-io/metal3-dev-env/pull/672#discussion_r644925590